static/js/chart/draw.ts

```diff
@@ -485,22 +485,32 @@ function drawLineChart(
   addYAxis(svg, width, yScale, unit);
 
   const legendText = dataGroups.map((dataGroup) =>
-    dataGroup.label ? dataGroup.label : "a"
+    dataGroup.label ? dataGroup.label : "A"
   );
   const colorFn = getColorFn(legendText);
 
   for (const dataGroup of dataGroups) {
     const dataset = dataGroup.value.map((dp) => {
       return [new Date(dp.label).getTime(), dp.value];
     });
+    const hasNullValues = dataset.reduce((accumulator, currentValue) => {
+      return accumulator || currentValue[1] === null;
+    }, false);
     const shouldAddDots = dataset.length < 12;
-    let line = d3
+
+    const line = d3
       .line()
+      .defined((d) => d[1] !== null)
       .x((d) => xScale(d[0]))
       .y((d) => yScale(d[1]));
 
-    if (shouldAddDots) {
-      line = line.curve(d3.curveMonotoneX);
+    if (hasNullValues) {
```

Comment:
This looks nice! Can you explain a bit more how this does the magic to use solid lines and dotted lines together? This is fancy d3 :)